labs/topologies/lab01_basic_frr/README.md

```diff
@@ -0,0 +1,396 @@
+## FRR Base Lab
+
+### Introduction
+
+This lab demonstrates basic principles of network automation using FRR devices. The main goal is demonstrating Annet's `Entire` generator type.
+
+Authors:
+- [Grigorii Macheev](https://github.com/gregory-mac),
+- [Vadim Volovik](https://github.com/vadvolo),
+- [Grigorii Solovev](https://github.com/gs1571)
+
+### Objectives
+
+- Understand main principles of writing Annet `Entire` generators
+
+### Topology
+
+![Lab Topology](./images/topology.png)
+
+### Environment
+
+- Netbox url: http://localhost:8000/
+- Netbox login/password: `annet/annet`
+- Device telnet and ssh login/password: `annet/annet`
+- Device mgmt addresses:
+   | Router | MGMT |
+   |:------:|:----|
+   | frr-r1 | `172.20.0.111` |
+   | frr-r2 | `172.20.0.112` |
+   | frr-r3 | `172.20.0.113` |
+
+
+### Generators
+
+This lab has single generator `./src/lab_generators/frr.py`.
+
+
+Unlike the `Partial` generators from the previous lab, which create and apply configuration line-by-line, the `Entire` type generates a whole configuration file in one go, which is then copied to the device. [More about Entire generators in annet documentation](https://annetutil.github.io/annet/main/usage/gen.html#entire).  
+FRR can be managed by `vtysh`, a Cisco-like CLI shell, but it also stores its configuration in a `/etc/frr/frr.conf` file.
+We can leverage this fact to manage the routing configuration in a server-like manner, and `Partial` generator will help us to prepare the configuration file.
```

Comment:
It's unclear how Partial generator helps us, because here we use only Entire. Did I miss something?

Reply:
and 'Entire'